sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/implementation/MemberNameConverterImpl.java

```diff
@@ -29,48 +28,12 @@
  */
 final class MemberNameConverterImpl implements MemberNameConverter {
     private static final ClientLogger LOGGER = new ClientLogger(MemberNameConverterImpl.class);
-
-    private static final String ACCESSOR_NAMING_STRATEGY =
-        "com.fasterxml.jackson.databind.introspect.AccessorNamingStrategy";
-    private static final String ACCESSOR_NAMING_STRATEGY_PROVIDER = ACCESSOR_NAMING_STRATEGY + "$Provider";
-    private static final MethodHandle GET_ACCESSOR_NAMING;
-    private static final MethodHandle FOR_POJO;
-    private static final MethodHandle FIND_NAME_FOR_IS_GETTER;
-    private static final MethodHandle FIND_NAME_FOR_REGULAR_GETTER;
-    static final boolean USE_REFLECTION_FOR_MEMBER_NAME;
+    static final boolean USE_JACKSON_212;
 
     private final ObjectMapper mapper;
 
     static {
-        MethodHandles.Lookup publicLookup = MethodHandles.publicLookup();
-
-        MethodHandle getAccessorNaming = null;
-        MethodHandle forPojo = null;
-        MethodHandle findNameForIsGetter = null;
-        MethodHandle findNameForRegularGetter = null;
-        boolean useReflectionForMemberName = false;
-
-        try {
-            Class<?> accessorNamingStrategyClass = Class.forName(ACCESSOR_NAMING_STRATEGY);
-            Class<?> accessorNamingStrategyProviderClass = Class.forName(ACCESSOR_NAMING_STRATEGY_PROVIDER);
-            getAccessorNaming = publicLookup.unreflect(MapperConfig.class.getDeclaredMethod("getAccessorNaming"));
-            forPojo = publicLookup.unreflect(accessorNamingStrategyProviderClass.getDeclaredMethod("forPOJO",
-                MapperConfig.class, AnnotatedClass.class));
-            findNameForIsGetter = publicLookup.unreflect(accessorNamingStrategyClass.getDeclaredMethod(
-                "findNameForIsGetter", AnnotatedMethod.class, String.class));
-            findNameForRegularGetter = publicLookup.unreflect(accessorNamingStrategyClass.getDeclaredMethod(
-                "findNameForRegularGetter", AnnotatedMethod.class, String.class));
-            useReflectionForMemberName = true;
-        } catch (LinkageError | ReflectiveOperationException ex) {
-            LOGGER.verbose("Failed to retrieve MethodHandles used to get naming strategy. Falling back to BeanUtils. "
-                + "Please consider updating your Jackson dependencies to at least version 2.12", ex);
-        }
-
-        GET_ACCESSOR_NAMING = getAccessorNaming;
-        FOR_POJO = forPojo;
-        FIND_NAME_FOR_IS_GETTER = findNameForIsGetter;
-        FIND_NAME_FOR_REGULAR_GETTER = findNameForRegularGetter;
-        USE_REFLECTION_FOR_MEMBER_NAME = useReflectionForMemberName;
+        USE_JACKSON_212 = PackageVersion.VERSION.getMinorVersion() >= 12;
```

Comment:
do we always know that `PackageVersion.VERSION` returns the version of the right package where coercion and other APIs we're guiding with are defined? We had a bunch of issues where jackson versions were inconsistent

Reply:
Yes, this is the PackageVersion for Jackson Databind which is where all these types are defined. I'm going to change this to an instance variable just in case this can somehow change during runtime.